### ISSUE TITLE
Wobbly edges

### ISSUE THREAD

**Issue by optimaltaskin** [open-mmlab/mmdetection#7494]
Hello,

I’m trying to train a credit card detector(card is held on forehead). I have a training set of 80 images, test set of 18 images and finally validation set of 21 images. I know my dataset is small for training but before moving forward I want to be sure whether I’m on the right track or not.

My problem is, when I train the detector, it’s able to find the card but especially when card is tilted, it detects edges wobbly. I’m sharing the two versions of the same image(one shows how I labelled the other shows how detection resulted).

![lab](https://user-images.githubusercontent.com/85254524/159427755-f74760e3-f653-4745-b26a-eeb6903d0b13.png)
![res](https://user-images.githubusercontent.com/85254524/159430029-16c864c7-33bb-413e-83a2-03ceede3e56d.png)

My config file is as below:
```
_base_ = '../mask_rcnn/mask_rcnn_r50_caffe_fpn_mstrain-poly_2x_coco.py'

model = dict(
    roi_head=dict(
        bbox_head=dict(num_classes=1),
        mask_head=dict(num_classes=1)))

dataset_type = 'COCODataset'
classes = ('card',)
data = dict(
    train=dict(
        img_prefix='data/base_set/training/',
        classes=classes,
        ann_file='data/base_set/training/training_labels.json'),
    val=dict(
        img_prefix='data/base_set/testing/',
        classes=classes,
        ann_file='data/base_set/testing/testing_labels.json'),
    test=dict(
        img_prefix='data/base_set/validation/',
        classes=classes,
        ann_file='data/base_set/validation/val_labels.json'),
)

load_from = 'checkpoints/mask_rcnn_r50_caffe_fpn_mstrain-poly_3x_coco_bbox_mAP-0.408__segm_mAP-0.37_20200504_163245-42aa3d00.pth'
```

I want to increase precision of how card is detected. I mean card edges should be detected with straight and more precise lines. I have been searching through the documents to find out where I'm doing wrong  but couldn't find anything right now. 
So I appreciate any recommendations for achieving my goal. Is my dataset too small? or should I use a different config file as base? etc.Thanks in advance.

**Comment by hhaAndroid** [comment]
The reason should be that the mask output size is too small.
```python
mask_roi_extractor=dict(
      type='SingleRoIExtractor',
      roi_layer=dict(type='RoIAlign', output_size=14, sampling_ratio=0), # 14
     out_channels=256,
     featmap_strides=[4, 8, 16, 32]),
```
You can consider modifying `output_size`  to 28 or change to [PointRend](https://github.com/open-mmlab/mmdetection/blob/master/configs/point_rend/point_rend_r50_caffe_fpn_mstrain_1x_coco.py)

**Comment by hhaAndroid** [comment]
If you use `pointrend`, you should modify the following configuration
```python
    # configs/point_rend/point_rend_r50_caffe_fpn_mstrain_1x_coco.py
    roi_head=dict(
        type='PointRendRoIHead',
        mask_roi_extractor=dict(
            type='GenericRoIExtractor',
            aggregation='concat',
            roi_layer=dict(
                _delete_=True, type='SimpleRoIAlign', output_size=14),
            out_channels=256,
            featmap_strides=[4]),
```

**Comment by optimaltaskin** [comment]
@hhaAndroid thanks for your recommendation. Now edges are much better:
![Screenshot from 2022-03-25 16-03-34](https://user-images.githubusercontent.com/85254524/160125982-9abaf3b5-b0ea-4db3-9c0c-be58f6cca1d5.png)

Yet I still need improvement in accuracy so will be working on that. Thanks :)
